heat/naive_bayes/gaussianNB.py

```diff
@@ -0,0 +1,555 @@
+    GaussianNB()
+    >>> print(clf.predict([[-0.8, -1]]))
+    [1]
+    >>> clf_pf = GaussianNB()
+    >>> clf_pf.partial_fit(X, Y, np.unique(Y))
+    GaussianNB()
+    >>> print(clf_pf.predict([[-0.8, -1]]))
+    [1]
+    """
+
+    def __init__(self, priors=None, var_smoothing=1e-9):
+        self.priors = priors
+        self.var_smoothing = var_smoothing
+
+    def fit(self, X, y, sample_weight=None):
+        """
+        Fit Gaussian Naive Bayes according to X, y
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : array-like, shape (n_samples,)
+            Target values.
+        sample_weight : array-like, shape (n_samples,), optional (default=None)
+            Weights applied to individual samples (1. for unweighted).
+
+        Returns
+        -------
+        self : object
+        """
+        # sanitize input
+        if not isinstance(X, ht.DNDarray):
+            raise ValueError("input needs to be a ht.DNDarray, but was {}".format(type(X)))
+        if not isinstance(y, ht.DNDarray):
+            raise ValueError("input needs to be a ht.DNDarray, but was {}".format(type(y)))
+        if y.numdims != 1:
+            raise ValueError("expected y to be a 1-D tensor, is {}-D".format(y.numdims))
+        if sample_weight is not None:
+            if not isinstance(sample_weight, ht.DNDarray):
+                raise ValueError(
+                    "sample_weight needs to be a ht.DNDarray, but was {}".format(
+                        type(sample_weight)
+                    )
+                )
+        classes = ht.unique(y, sorted=True)
+        if classes.split is not None:
+            classes = ht.resplit(classes, axis=None)
+
+        return self.__partial_fit(X, y, classes, _refit=True, sample_weight=sample_weight)
+
+    def __check_partial_fit_first_call(self, classes=None):
+        """
+        Private helper function for factorizing common classes param logic
+        Estimators that implement the ``partial_fit`` API need to be provided with
+        the list of possible classes at the first call to partial_fit.
+        Subsequent calls to partial_fit should check that ``classes`` is still
+        consistent with a previous value of ``clf.classes_`` when provided.
+        This function returns True if it detects that this was the first call to
+        ``partial_fit`` on ``clf``. In that case the ``classes_`` attribute is also
+        set on ``clf``.
+        """
+        if getattr(self, "classes_", None) is None and classes is None:
+            raise ValueError("classes must be passed on the first call " "to partial_fit.")
+
+        elif classes is not None:
+            unique_labels = classes
+            if getattr(self, "classes_", None) is not None:
+                if not ht.equal(self.classes_, unique_labels):
```

Comment:
could these two be combined? or maybe it could be something like this
```python
if getattr(self, "classes_", None) is None:
    self.classes_ = unique_labels
    return True
if not ht.equal(self.classes_, unique_labels):
    raise ValueError("...")
return False
```

Reply:
thoughts?

Reply:
Sorry I missed  your first comment, you're right, I've updated according to your suggestion.